core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/first.kt

```diff
@@ -62,26 +173,209 @@ public inline fun <T> DataFrame<T>.firstOrNull(predicate: RowFilter<T>): DataRow
+ * ### Example
+ * ```kotlin
+ * // Select the first employee from each group formed by the job title
+ * employees.groupBy { jobTitle }.first()
+ * ```
+ *
+ * @param T The type of the values in the [GroupBy].
+ * @param G The type of the groups in the [GroupBy].
+ *
+ * @return A [ReducedGroupBy] containing the first row from each group.
+ */
 @Interpretable("GroupByReducePredicate")
 public fun <T, G> GroupBy<T, G>.first(): ReducedGroupBy<T, G> = reduce { firstOrNull() }
 
+/**
+ * Selects from each group of the given [GroupBy] the first row satisfying the given [predicate],
+ * and returns a [ReducedGroupBy] containing these rows (one row per group,
+ * each row is the first row in its group that satisfies the [predicate]).
+ *
+ * If the group in [GroupBy] contains no matching rows,
+ * the corresponding row in [ReducedGroupBy] will contain `null` values for all columns in the group.
+ *
+ * {@include [RowFilterDescription]}
+ *
+ * @include [SelectingColumns.ColumnGroupsAndNestedColumnsMention]
+ *
+ * ### Example
+ * ```kotlin
+ * // Select the first employee older than 25 from each group formed by the job title
+ * employees.groupBy { jobTitle }.first { age > 25 }
+ * ```
+ *
+ * @param T The type of the values in the [GroupBy].
+ * @param G The type of the groups in the [GroupBy].
+ * @param predicate A lambda expression used to select a value
+ * that satisfies a condition specified in this expression.
+ * This predicate takes a value from the [GroupBy] as an input
+ * and returns `true` if the value satisfies the condition or `false` otherwise.
+ *
+ * @return A [ReducedGroupBy] containing the first row matching the [predicate]
+ * (or a row with `null` values, except values in the column with the grouping key), from each group.
+ */
 @Interpretable("GroupByReducePredicate")
 public fun <T, G> GroupBy<T, G>.first(predicate: RowFilter<G>): ReducedGroupBy<T, G> = reduce { firstOrNull(predicate) }
 
 // endregion
 
 // region Pivot
 
+/**
+ * Reduces this [Pivot] by selecting the first row from each group.
+ *
+ * Returns a [ReducedPivot] where:
+ * - each column corresponds to a [pivot] group — if multiple pivot keys were used,
+ *   the result will contain column groups for each pivot key, with columns inside
+ *   corresponding to the values of that key;
+ * - each value contains the first row from that group.
+ *
+ * The original [Pivot] column structure is preserved.
+ * If the [Pivot] was created using multiple or nested keys
+ * (e.g., via [and][PivotDsl.and] or [then][PivotDsl.then]),
+ * the structure remains unchanged — only the contents of each group
+ * are replaced with the first row from that group.
+ *
+ * Equivalent to `reduce { firstOrNull() }`.
+ *
+ * See also:
+ * - [pivot];
+ * - common [reduce][Pivot.reduce].
+ *
+ * ### Example
+ * ```kotlin
+ * // Select the first row for each city.
+ * // Returns a ReducedPivot with one column per city and the first row from the group in each column.
+ * df.pivot { city }.first()
```

Comment:
Please see if you can come up with representative example. Like, in what situation you'd use this function? What df typically it will be and what ideas one can draw from the result? Will be good if example can convey this